src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -524,6 +524,27 @@ public override BoundNode VisitYieldReturnStatement(BoundYieldReturnStatement no
             return UpdateStatement(builder, node.Update(expression));
         }
 
+        public override BoundNode VisitCatchBlock(BoundCatchBlock node)
```

Comment:
>BoundCatchBlock [](start = 50, length = 15)

Is it worth reviewing the list of bound nodes to see if any other should be handled by the SpillSequenceSpiller?
Also, I'm wondering how we can help catch this issue as we add new bound nodes in the future. Maybe add a note for spill sequences in test plan, or we could document all the visitors to consider in the codebase?

Reply:
Yes, good idea. Reviewing all of the constructs that *could* be affected (by finding all the places that a `BoundSpillSequence` is created):
- `await` is forbidden in a catch filter.
- `stackalloc` is forbidden in a catch filter.
- `switch` and complex-enough `is-pattern` operators lower the same way, and are fixed by this PR.
I'm adding a note to the test plan.


---
In reply to: [456718535](https://github.com/dotnet/roslyn/pull/46054#discussion_r456718535) [](ancestors = 456718535)